tests/test_bundler.py

```diff
@@ -0,0 +1,445 @@
+# test_bundler.py
+"""Unit tests for lta/bundler.py."""
+
+from asyncio import Future
+from unittest.mock import call, MagicMock
+import pytest  # type: ignore
+import requests
+from tornado.web import HTTPError  # type: ignore
+from unittest.mock import patch, mock_open
+
+from lta.bundler import main, patch_status_heartbeat, Bundler, status_loop, work_loop
+
+
+class AsyncMock(MagicMock):
+    """
+    AsyncMock is the async version of a MagicMock.
+
+    We use this class in place of MagicMock when we want to mock
+    asynchronous callables.
+
+    Source: https://stackoverflow.com/a/32498408
+    """
+
+    async def __call__(self, *args, **kwargs):
+        """Allow MagicMock to work its magic too."""
+        return super(AsyncMock, self).__call__(*args, **kwargs)
+
+
+class ObjectLiteral:
+    """
+    ObjectLiteral transforms named arguments into object attributes.
+
+    This is useful for creating object literals to be used as return
+    values from mocked API calls.
+
+    Source: https://stackoverflow.com/a/3335732
+    """
+
+    def __init__(self, **kwds):
+        """Add attributes to ourself with the provided named arguments."""
+        self.__dict__.update(kwds)
```

Comment:
These two helper classes have appeared a few times, which means they probably need to be in a util file for testing.

Reply:
These got refactored out to `tests/test_util.py` and the unit tests for Picker, Bundler, and monitoring were refactored to use them.